playground/src/pages/CodingEnvironment.tsx

```diff
@@ -15,6 +16,98 @@ async function getExample(example: string) {
 
 function CodingEnvironment() {
 
+   /**
+  * Helper function to check if input contains only numbers.
+  */ 
+   function checkInputField(jsonField: JSON, key: string) {
+
+    const jsonInput = jsonField[key as keyof typeof jsonField];
+
+    if (!Array.isArray(jsonInput)) {
+
+      const errorMessage = `If you use ${key},
+      it must be an array of numbers.`;
+
+      toast.error(errorMessage);
+
+      setOutput(errorMessage);
+
+      return false;
+
+    }
+
+    if (Object.values(jsonInput).length === 0 || Object.values(jsonInput).some(isNaN)) {
+
+      const errorMessage = `If you use ${key}, 
+      it must contain at least one number, 
+      and it can only contain numbers.`;
+
+      toast.error(errorMessage);
+
+      setOutput(errorMessage);
+
+      return false;
+    }
+
+    return true;
+
+    }
+
+
+  /**
+  * We check the inputs and return true or false. We allow: 
+  * - an empty input, and
+  * - a valid JSON object containing stack_init or advice_tape (or both) 
+  * if the values are numbers
+  */ 
+  function checkInputs(jsonString: string) {
+
+    if (jsonString === "") {
+      return true;
+    }
+
+    if (!jsonString.includes("stack_init") && !jsonString.includes("advice_tape")) {
```

Comment:
instead of checking to see if `stack_init` is a substring of `jsonString`. Instead it should check if `stack_init` is a key of the parsing result. (same for `advice_tape`)

Reply:
got it, is changed